Source/Public/New-ADOPSElasticpool.ps1

```diff
@@ -0,0 +1,44 @@
+function New-ADOPSElasticPool {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [string]$PoolName,
+        
+        # Request body found at https://docs.microsoft.com/en-us/rest/api/azure/devops/distributedtask/elasticpools/create?view=azure-devops-rest-7.1
+        [Parameter(Mandatory)]
+        [string]
+        $Body,
+
+        [string]$Organization,
+        
+        [Parameter()]
+        [guid]
+        $ProjectId,
+
+        [Parameter()]
+        [boolean]
+        $AuthorizeAllPipelines = $false,
+
+        [Parameter()]
+        [boolean]
+        $AutoProvisionProjectPools = $false
+
+    )
+
+    if (-not [string]::IsNullOrEmpty($Organization)) {
+        $Org = GetADOPSHeader -Organization $Organization
+    } else {
+        $Org = GetADOPSHeader
+        $Organization = $Org['Organization']
+    }
+
+    if ($PSBoundParameters.ContainsKey('ProjectId')) {
+        $Uri = "https://dev.azure.com/$Organization/_apis/distributedtask/elasticpools?poolName=$PoolName`&authorizeAllPipelines=$AuthorizeAllPipelines`&autoProvisionProjectPools=$AutoProvisionProjectPools&projectId={projectId}&api-version=7.1-preview.1"
+    } else {
+        $Uri = "https://dev.azure.com/$Organization/_apis/distributedtask/elasticpools?poolName=$PoolName`&authorizeAllPipelines=$AuthorizeAllPipelines`&autoProvisionProjectPools=$AutoProvisionProjectPools&api-version=7.1-preview.1"
+    }
+    
+    $Method = 'POST'
+    $ElasticPoolInfo = InvokeADOPSRestMethod -Uri $Uri -Method $Method -Organization $Organization -Body $Body
```

Comment:
Can we include a default body? There are some parameters I would like to have settable, but most can be just "standard" for now. The option would be to also add a command like `New-ADOPSElasticPoolObject` or something that we could use to create a body or object we could then use as parameter value in this command.

Reply:
I've added a new function called New-ADOPSElasticPoolObject
Let's discuss and see how we can best implement it!